hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMPerformanceMetrics.java

```diff
@@ -119,6 +119,21 @@ public static void unregister() {
   @Metric(about = "resolveBucketLink latency in listKeys")
   private MutableRate listKeysResolveBucketLatencyNs;
 
+  @Metric(about = "deleteKeyFailure latency in nano seconds")
+  private MutableRate deleteKeyFailureLatencyNs;
+
+  @Metric(about = "deleteKeySuccess latency in nano seconds")
+  private MutableRate deleteKeySuccessLatencyNs;
```

Comment:
For `deleteKeyFailure` and `deleteKeySuccess` related metrics, do you think it would be more readable if we did a combined stat update for both the call counters as well as their individual call latencies? Something such as the following for success and failure results, similar to what has been implemented for `SCMPerformanceMetrics`/`S3GatewayMetrics`:

```
  public void updateDeleteKeySuccessStats(long startNanos) {
    deleteKeySuccess.incr();
    deleteKeySuccessLatencyNs.add(Time.monotonicNowNanos() - startNanos);
  }

  public void updateDeleteKeyFailureStats(long startNanos) {
    deleteKeyFailure.incr();
    deleteKeyFailureLatencyNs.add(Time.monotonicNowNanos() - startNanos);
  }
```

As it looks like we are capturing a `MutableCounterLong` metric already for the `numKeyDeleteFails` in both `OMKeyDeleteRequest` and `OMKeysDeleteRequest` separately.

Reply:
We are creating these metrics (latency related) in OMPerformance class for tracking performance and those counter metrics are available on another metric class OMMetric. That's why I have done it in this way.